packages/graphql/src/graphql.ts

```diff
@@ -69,8 +48,10 @@ function generateName(type: types.Type, defaultName: string | undefined): string
 type InternalData = {
   // A set of all the types that have already been explored
   inspectedTypes: Set<types.Type>
-  // A map from <explored type> to already generated output type
-  knownTypes: Map<types.Type, GraphQLOutputType>
+  // A map from <explored type> to already generated output type(s)
+  knownOutputTypes: Map<types.Type, GraphQLOutputType>
+  // A map from <explored type> to already generated input type(s)
+  knownInputTypes: Map<types.Type, GraphQLInputType>
```

Comment:
Added this so that I can pass the same data structure to both and intermediate results can be cached for both input/output types